sympy/solvers/ode/ode.py

```diff
@@ -2259,6 +2262,23 @@ def odesimp(ode, eq, func, hint):
        \2*x /
 
     """
+    if isinstance(eq, BooleanFalse):
+        return None
+    if hint == '2nd_autonomous_nth':
```

Comment:
This `if` condition is only implemented for the case when `n` is a symbol in the ode passed. Otherwise, if `n` is an arbitrary number, this condition is not useful to us. This condition ensures to get a list of solutions such that we can apply `subs` in them.
Why we need this condition- [reason](https://github.com/sympy/sympy/pull/20845#issuecomment-780678266).
The main issue is with `solve`,
```
In [1]: solve(x**n-1,x)
Out[1]: [1]
```
Here what if `n` was even integer, or maybe any other odd integer except for `n=1`, However it works fine for the particular values when `n` is provided
Kindly share your opinion.
Thanks.

Reply:
To me that just seems like a bug in `solve`. I would rather not workaround it here but instead just open an issue describing the problem. If `dsolve` doesn't give the complete answer in this case then that can be improved later when solve is improved or if a different solver is used instead.

Reply:
Sure, [solve.](https://github.com/sympy/sympy/issues/21014)
Thanks.

Reply:
I think this case should be removed from `odesimp`. The function should either be fixed in general or left alone. There should not be special checks for particular solvers.

Reply:
Indeed, will try to fix for `solve`, probably that will handle the issues and we can omit the `if` condition.
Thanks for replying.